cedar-policy-core/src/est/scope_constraints.rs

```diff
@@ -272,6 +272,25 @@ impl PrincipalConstraint {
             }
         }
     }
+
+    /// Returns true if this constraint has a slot.
+    pub fn has_slot(&self) -> bool {
+        match self {
+            PrincipalConstraint::All => false,
+            PrincipalConstraint::Eq(EqConstraint::Entity { entity: _ }) => false,
+            PrincipalConstraint::Eq(EqConstraint::Slot { slot: _ }) => true,
```

Comment:
Convenient syntax sugar:
```suggestion
            PrincipalConstraint::Eq(EqConstraint::Entity { .. }) => false,
            PrincipalConstraint::Eq(EqConstraint::Slot { .. }) => true,
```